tests/suites/test_suite_mdx.function

```diff
@@ -2,6 +2,7 @@
 #include "mbedtls/md2.h"
 #include "mbedtls/md4.h"
 #include "mbedtls/md5.h"
+#include "mbedtls/sm3.h"
```

Comment:
Please create a `test_suite_sm3` that tests the low-level `mbedtls_sm3_xxx` interface directly, on the model of `test_suite_mdx` and `test_suite_shax`. `sm3_selftest` should go there instead, and there should be test vectors as well.

`test_suite_sm3` needs to be declared in `tests/CMakeLists.txt` (`tests/Makefile` will pick it up automatically).